src/components/fields/ArrayField.js

```diff
@@ -161,19 +161,6 @@ class ArrayField extends Component {
 
   constructor(props) {
     super(props);
-    this.state = this.getStateFromProps(props);
-  }
```

Comment:
The constructor is now unnecessary. Also that means we don't have local state anymore, which means we should use a functional stateless component instead of a component class. Which is so nice.